src/icos_config/taklimakan.js

```diff
@@ -0,0 +1,80 @@
+      lastTransactionBlockNumber: null, // follow last block
+      countTransactions: true,
+    },
+  },
+  icoParameters: {
+    cap: async (web3, icoContract, tokenContract) => {
+      const tklnCap = await toPromise(icoContract.saleGoal)();
+      return [`${formatNumber(convertWeb3Value(tklnCap))} TKLN`];
+    },
+    startDate: async (web3, icoContract) => {
+      const startDate = await toPromise(icoContract.saleStartTimestamp)();
+      return convertWeb3Value(startDate.valueOf(), 'timestamp').formatDate();
+    },
+    endDate: async (web3, icoContract) => {
+      const endDate = await toPromise(icoContract.saleStopTimestamp)();
+      return convertWeb3Value(endDate.valueOf(), 'timestamp').formatDate();
+    },
+    status: async (web3, icoContract) => {
+      const now = Math.floor(new Date().getTime() / 1000);
+      const startDateICO = await toPromise(icoContract.saleStartTimestamp)();
+      const endDateICO = await toPromise(icoContract.saleStopTimestamp)();
+      const goalReached = await toPromise(icoContract.goalReached)();
+      if (now < startDateICO) {
+        return 'not started';
+      } else if (now >= startDateICO && now < endDateICO) {
+        if (goalReached) {
+          return 'successful';
+        }
+        return 'in progress';
+      }
+      if (goalReached) {
+        return 'successful';
+      }
+      return 'failed';
+    },
+  },
+  matrix: {
+    q1: { answer: true, comment: 'Yes, the contract name is TaklimakanCrowdsale' },
+    q2: { answer: true, comment: 'Source code is available' },
+    q3: { answer: true, comment: '' },
+    q4: { answer: true, comment: '' },
+    q5: { answer: true, comment: '' },
+    q6: { answer: true, comment: 'Token rate is provided via TokenSale Event as custom argument: tokensPerEther' },
+    q7: { answer: true, comment: 'ETH is sent to smart contract and cannot be withdrawn by owner until the ICO ends (either by reaching maximum cap or by reaching ICO close date)' },
+    q8: { answer: true, comment: 'n/a: Taklimakan is only using ETH' },
+    q9: { answer: true, comment: 'n/a: Taklimakan is only using ETH' },
+    q10: { answer: true, comment: '' },
+    q11: { answer: true, comment: '' },
+    q12: { answer: true, comment: 'Both price and bonus are controlled by smart contract, see processPayment function in source code.' },
+    q13: { answer: true, comment: 'Check in the source code how processPayment function uses SaleStartTimestamp variable that stores ICO start date. SaleStartTimestamp cannot change after contract is deployed.' },
+    q14: { answer: true, comment: 'Check in the source code how processPayment function uses SaleStopTimestamp variable that stores ICO end date. SaleStopTimestamp cannot change after contract is deployed. Also, general sale ends when the maximum cap is reached, which is also controlled by this smart contract.' },
+  },
+  decimals: 18,
+  addedBy: 'gzaytsev',
+  dateAdded: '23-11-2017',
```

Comment:
you can provide current date here, so you are listed more to the top. like 11-12-2017

Reply:
Thank you!